hw/ip/otbn/dv/model/otbn_trace_checker.cc

```diff
@@ -50,89 +50,84 @@ void OtbnTraceChecker::AcceptTraceString(const std::string &trace,
   done_ = false;
   OtbnTraceEntry trace_entry;
   trace_entry.from_rtl_trace(trace);
-  if (trace_entry.empty()) {
-    std::cerr << "ERROR: Invalid RTL trace entry with empty header:\n";
+  if (trace_entry.trace_type() == OtbnTraceEntry::Invalid) {
+    std::cerr << "ERROR: Invalid RTL trace entry with invalid header:\n";
     trace_entry.print("  ", std::cerr);
     seen_err_ = true;
     return;
   }
 
-  // Unless something has gone very wrong, trace_entry.hdr_ will start with 'S'
-  // (stall) or 'E' (execute). We want to coalesce entries for an instruction
-  // here to avoid the ISS needing to figure out what write happens when on a
-  // multi-cycle instruction.
+  // Check we don't already have a pending RTL final entry
+  if (rtl_pending_) {
+    std::cerr
+        << ("ERROR: Two back-to-back RTL "
+            "trace entries with no ISS entry.\n"
+            "  First RTL entry was:\n");
+    rtl_entry_.print("    ", std::cerr);
+    std::cerr << "  Second RTL entry was:\n";
+    trace_entry.print("    ", std::cerr);
+    seen_err_ = true;
+    return;
+  }
+
+  // At this point, we know that the header is of type S, E, X or Y. We want to
```

Comment:
Paging Elon Musk....

Reply:
Oh my god. I completely missed that. :facepalm:

Reply:
Note there's a few places to re-order

Reply:
You're right. Since the X and Y were pretty arbitrary, I've changed them to U and V.